<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>AtCoder Contest Helper</string>
  </property>
  <property name="windowIcon">
   <iconset resource="res.qrc">
    <normaloff>:/images/icons/icon.png</normaloff>:/images/icons/icon.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="contestTab">
       <attribute name="title">
        <string>比赛</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QLabel" name="labContestId">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>比赛 ID:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="contestIdEdit"/>
          </item>
          <item>
           <widget class="QPushButton" name="btnFetchProblems">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>获取题目</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTableWidget" name="contestTable">
          <property name="editTriggers">
           <set>QAbstractItemView::NoEditTriggers</set>
          </property>
          <property name="selectionMode">
           <enum>QAbstractItemView::SingleSelection</enum>
          </property>
          <property name="columnCount">
           <number>4</number>
          </property>
          <column/>
          <column/>
          <column/>
          <column/>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="problemTab">
       <attribute name="title">
        <string>题目</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="labContestId_2">
            <property name="text">
             <string>比赛 ID:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="contestIdEdit_2"/>
          </item>
          <item>
           <widget class="QLabel" name="labProblemId_2">
            <property name="text">
             <string>题目 ID:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="problemIdEdit_2"/>
          </item>
          <item>
           <widget class="QPushButton" name="btnFetchProblem">
            <property name="text">
             <string>获取题面</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btnSaveAsMarkdown">
            <property name="text">
             <string>保存为Markdown</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTextBrowser" name="problemBrowser"/>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_6">
          <item>
           <widget class="QPushButton" name="btnJumpToContest">
            <property name="text">
             <string>转到比赛</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btnSubmitCode">
            <property name="text">
             <string>提交代码</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="submitTab">
       <attribute name="title">
        <string>提交</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QLabel" name="labContestId_3">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>比赛 ID:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="contestIdEdit_3"/>
          </item>
          <item>
           <widget class="QLabel" name="labProblemId_3">
            <property name="text">
             <string>题目 ID:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="problemIdEdit_3"/>
          </item>
          <item>
           <widget class="QPushButton" name="btnGetLanguages">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>获取可用语言</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QLabel" name="labLanguage">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>提交语言:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="comboBox_language">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btnSubmit">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>提交</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLabel" name="labSubmissionCode">
          <property name="text">
           <string>代码:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPlainTextEdit" name="codeEdit">
          <property name="font">
           <font>
            <family>Consolas</family>
            <pointsize>12</pointsize>
           </font>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="settingsTab">
       <attribute name="title">
        <string>设置</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <widget class="QLabel" name="labCurrentUser">
            <property name="text">
             <string>当前用户：未登录</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btnLogin">
            <property name="text">
             <string>登录</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLabel" name="labSettingsPath">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>配置文件保存路径：</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actContest_Fetch">
   <property name="text">
    <string>获取比赛题目</string>
   </property>
   <property name="toolTip">
    <string>获取比赛题目</string>
   </property>
  </action>
  <action name="actProblem_Fetch">
   <property name="text">
    <string>获取题面</string>
   </property>
  </action>
  <action name="actSubmit_getLanguages">
   <property name="text">
    <string>获取可用语言</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="res.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>btnFetchProblems</sender>
   <signal>clicked()</signal>
   <receiver>actContest_Fetch</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>739</x>
     <y>76</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>contestIdEdit</sender>
   <signal>returnPressed()</signal>
   <receiver>actContest_Fetch</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>387</x>
     <y>76</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>problemIdEdit_2</sender>
   <signal>returnPressed()</signal>
   <receiver>actProblem_Fetch</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>558</x>
     <y>76</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnFetchProblem</sender>
   <signal>clicked()</signal>
   <receiver>actProblem_Fetch</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>739</x>
     <y>76</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnGetLanguages</sender>
   <signal>clicked()</signal>
   <receiver>actSubmit_getLanguages</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>736</x>
     <y>76</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>problemIdEdit_3</sender>
   <signal>returnPressed()</signal>
   <receiver>actSubmit_getLanguages</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>549</x>
     <y>76</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
